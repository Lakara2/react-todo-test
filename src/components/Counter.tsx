import { useState } from "react";

export function Counter () {
    const [count, setCount] = useState<number>(0);
    
    function addition () {
        setCount ( count + 1 )
    }

    return(
        <div className='container'>
            <div className='row'>
                <h1 className='col-6 form-control' data-testid={"bryan"}>{count}</h1>
                <button
                    className='col-3 btn-danger btn'
                    type="button"
                    onClick={addition}
                    >Add
                </button>
            </div>
        </div>
    );
}