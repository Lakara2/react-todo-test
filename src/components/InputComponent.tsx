import * as React from 'react';

export const InputComponent: React.FC<{
  AddOnTodo: (e:any) => any
}> = (props) => {
  const {AddOnTodo} = props;
  
  const [todo, setTodo] = React.useState("");

  const onTodo = (e:any):void => {
    if(e.target.id === "todo") {
        setTodo(e.target.value);
    }
  }

  const OnEnter=(e:any)=>{
    if(e.key==="Enter"){
      AddOnTodo(todo);
      setTodo("");
    }
  }
    return (
    <input type={"text"} className={"form-control mt-2 mb-2"}
     id="todo" value={todo} onChange={onTodo} onKeyDown={OnEnter}/>        
    )
}