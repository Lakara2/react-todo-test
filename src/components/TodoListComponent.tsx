import { useState } from "react";

export const TodoListComponent: React.FC<{
}> = (props) => {
  const [todoList,setTodoList] = useState([""]);
  const [todoDone, setTodoDone] = useState([""]);
  
  const onChecking=(index:number)=>{
    const item1=todoList[index];
    setTodoDone([...todoDone,item1]);
    todoList.splice(index, 1)
    setTodoList([...todoList])
  }
return (
    <div className='form-control' >
      <ul>
        {todoList.map((e,index)=>(
        <li key={index}>
          {e}
      <input type="checkbox" id='checkbox' className='chek' checked={false} 
      onChange={()=> onChecking(index)}/>
        </li>
        ))}
      </ul>
    </div>)
}