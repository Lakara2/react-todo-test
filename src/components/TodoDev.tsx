//TP: UNIT TEST - STD21062
import '../bootstrap.css'
import '../App.css';
import { useState } from 'react';
import { Counter } from './Counter';
import { InputComponent } from './InputComponent';
export function TodoDev() {

  //const [todo, setTodo] = useState("");
  //list
  const [todoList, setTodoList] = useState([""]);
  const [todoDone, setTodoDone] = useState([""]);

  const AddOnTodo =(todo: string)=>{
    if(todo!==""){
      setTodoList([...todoList,todo])}
  }
/*
  const OnEnter=(e:any)=>{
    if(e.key==="Enter"){
      AddOnTodo(todo);
      setTodo("");
    }
  }

  const onTodo = (e:any):void => {
    if(e.target.id === "todo") {
        setTodo(e.target.value);
    }
  }*/
            /*<input type={"text"} className={"form-control mt-2 mb-2"}
             id="todo" value={todo} onChange={onTodo} onKeyDown={OnEnter}/>*/
  const onChecking=(index:number)=>{
    const item1=todoList[index];
    setTodoDone([...todoDone,item1]);
    todoList.splice(index, 1)
    setTodoList([...todoList])
  }

  return (
    <>
      <div className='container'> 
        <div className='row'>
  
          <div className='col-5 m-3 tests bg-primary1'> <h1 className="m-2">Todo</h1> 
          <InputComponent AddOnTodo={AddOnTodo}/>
            <div className='form-control' >
              <ul>
                {todoList.map((e,index)=>(
                <li key={index}>
                  {e}
              <input type="checkbox" id='checkbox' className='chek' checked={false}
              onChange={()=>onChecking(index)}/>
                </li>
                ))}
              </ul>
            </div>
          </div>
  
          <div className='col-5 m-3 tests bg-success1'> 
            <h1 className="m-2">Done </h1>
            <div>
                {todoDone.map((e,index)=>(
              <div key={index}>{e}
              </div>))}
            </div>
          </div>

          <Counter/>
          
        </div>
      </div>
    </>
    );
}