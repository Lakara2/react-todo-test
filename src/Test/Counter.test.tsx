import React from 'react';
import TestRenderer from 'react-test-renderer';
import { isFunctionDeclaration } from 'typescript';
import { Counter } from "../components/Counter";

describe('Counter', () => {
    it('renderer correctly', () => {
      const renderer = TestRenderer.create(<Counter />);
      const json = renderer.toJSON();
      expect(json).toMatchSnapshot();
    });

    it('increments the count when the button is clicked', async () => {
      //creer le conteur
        const renderer = TestRenderer.create(<Counter/>);
        //instancier
        const instance = renderer.root;
        //declarer la fonctinon
        const addition = isFunctionDeclaration.call;
        //declarer et instancier le boutton
        const button = await instance.findByType('button');
        //dire ce que le boutton doit faire
        const go = button.props.onClick(addition);
        //expected au format json
        const json = renderer.toJSON();
        //snapshot
        expect(json).toMatchSnapshot();
        //expected(objet retourner dans tsx) vs reality(objet retourner dans le test)
        expect(json).toBe(go);
      });
  });