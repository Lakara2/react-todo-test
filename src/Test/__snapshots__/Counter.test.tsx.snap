// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter increments the count when the button is clicked 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1
      className="col-6 form-control"
    >
      Count : 
      1
    </h1>
    <button
      className="col-3 btn-danger btn"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
</div>
`;

exports[`Counter renderer correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1
      className="col-6 form-control"
    >
      Count : 
      0
    </h1>
    <button
      className="col-3 btn-danger btn"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
</div>
`;
